neqo-transport/src/connection.rs

```diff
@@ -1620,8 +1652,21 @@ impl Connection {
                 stateless_reset_token,
                 ..
             } => {
-                self.connection_ids
-                    .insert(sequence_number, (connection_id, stateless_reset_token));
+                // If adding an additional active connection id sends us over the limit,
+                // return a ConnectionIdLimitError.
+                if self.connection_ids.len() + 1
+                    > self
+                        .tps
+                        .borrow()
+                        .remote()
+                        .get_integer(tp_constants::ACTIVE_CONNECTION_ID_LIMIT)
+                        as usize
```

Comment:
We should be avoiding the cast in favour of `try_from()`.

Also, this would likely benefit from a temporary variable.

Reply:
I definitely agree with both. In fact, I had it as a temporary variable and then switched it back. Thanks for confirming my initial gut call on that. 

As for the `try_from`, I think that's a great idea. However, what I realized is what _should_ be done is to upcast from `usize` into `u64` and do the comparison on that. There is no possible data loss in that case. Does that sound okay?

Reply:
Doesn't that mean you would just be `try_from`ing the other side of the comparison? Either is fine, this is an extreme corner case and all this is only to please the compiler.